cmd/kubeadm/app/componentconfigs/defaults.go

```diff
@@ -127,3 +189,7 @@ func DefaultKubeletConfiguration(internalcfg *kubeadmapi.ClusterConfiguration) {
 	// Go back to the internal version
 	Scheme.Convert(externalkubeletcfg, internalcfg.ComponentConfigs.Kubelet, nil)
 }
+
+func warnDefaultComponentConfig(paramName string, componentName string, defaultValue interface{}) {
+	klog.Warningf("The recommended value for %s for %s is %v", paramName, componentName, defaultValue)
+}
```

Comment:
i think it makes more sense to have the following utility function:

```go
// warnDefaultComponentConfigValue prints a warning if the user modified a field in a certain
// CompomentConfig from the default recommended value in kubeadm.
func warnDefaultComponentConfigValue(componentConfigKind, paramName string, defaultValue, userValue interface{}) {
	klog.Warningf("The recommended value for %q in %q is: %v; the provided value is: %v",
		paramName, componentConfigKind, defaultValue, userValue)
}
```

- added comment on top of the function.
- renamed to add `Value` in function name.
- added userValue (this will make it easier for the user to find their value).
- use componentConfigKind; this means that the callers have to pass e.g. `{externalproxycfg| externalkubeletcfg}.Kind`
this solves the potential problem if the user feeds a value from a certain Kind, but kubeadm supports multiple different Kinds for a certain component. still does not solve the problem if a certain filed name is available multiple times in different nestings under the same Kind.
- adjusted warning to be on two lines.

Reply:
Sure, I will fix this.

Reply:
one final missing item:
> use componentConfigKind; this means that the callers have to pass e.g. {externalproxycfg| externalkubeletcfg}.Kind

so passing the above instead of `constants.Kubelet` and `constants.KubeProxy`.

[1]

Reply:
Got it